### PR TITLE
Fix some dav sharing ux

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Comment by PVince81** [comment]
Code looks good :+1:

File: apps/dav/lib/dav/sharing/backend.php [hunk 1]
```diff
@@ -43,8 +44,9 @@ class Backend {
 	 *
 	 * @param IDBConnection $db
 	 */
```
Comment (LukasReschke) [inline apps/dav/lib/dav/sharing/backend.php]: PHPDoc is lying now

File: apps/dav/lib/dav/sharing/backend.php [hunk 2]
```diff
@@ -153,9 +155,10 @@ public function getShares($resourceId) {
 
 		$shares = [];
 		while($row = $result->fetch()) {
+			$p = $this->principalBackend->getPrincipalByPath($row['principaluri']);
 			$shares[]= [
 				'href' => "principal:${row['principaluri']}",
-//				'commonName' => isset($p['{DAV:}displayname']) ? $p['{DAV:}displayname'] : '',
```
Comment (LukasReschke) [inline apps/dav/lib/dav/sharing/backend.php]: ?

File: apps/dav/lib/dav/sharing/backend.php [hunk 3]
```diff
@@ -43,8 +44,9 @@ class Backend {
 	 *
 	 * @param IDBConnection $db
 	 */
-	public function __construct(IDBConnection $db, $resourceType) {
+	public function __construct(IDBConnection $db, Principal $principalBackend, $resourceType) {
 		$this->db = $db;
+		$this->principalBackend = $principalBackend;
```
Comment (LukasReschke) [inline apps/dav/lib/dav/sharing/backend.php]: Missing member variables

**Comment by LukasReschke** [comment]
Will fix my remarks. Other stuff looks good.

**Comment by LukasReschke** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
THX

**Pull request merged**